mapping/document.go

```diff
@@ -534,33 +534,33 @@ func (dm *DocumentMapping) processProperty(property interface{}, path []string,
 			dm.walkDocument(property, path, indexes, context)
 		}
 	case reflect.Map, reflect.Slice:
-		var isPropertyVector bool
-		var isPropertyVectorInitialized bool
-		if subDocMapping != nil {
+		walkDocument := false
+		if subDocMapping != nil && len(subDocMapping.Fields) != 0 {
 			for _, fieldMapping := range subDocMapping.Fields {
 				switch fieldMapping.Type {
 				case "vector":
-					processed := fieldMapping.processVector(property, pathString, path,
+					fieldMapping.processVector(property, pathString, path,
 						indexes, context)
-					if !isPropertyVectorInitialized {
-						isPropertyVector = processed
-						isPropertyVectorInitialized = true
-					} else {
-						isPropertyVector = isPropertyVector && processed
-					}
```

Comment:
@Likith101 I believe @CascadingRadium  added this and my memory on this is somewhat hazy - it seems we did this to avoid having to process a vector field multiple times, I like your code proposal here which is simple and easy to follow - just want to make sure we're not inadvertently dropping any optimization here.

Reply:
I collaborated with @CascadingRadium on this and we made sure to preserve any existing behaviour that was present before vectors and only remove the double processing for vector fields.